extensions/azurecore/src/account-provider/utils/msalCachePlugin.ts

```diff
@@ -61,4 +81,28 @@ export class MsalCachePluginProvider {
 			afterCacheAccess,
 		};
 	}
+
+	async WaitAndLock(lockFilePath: string) {
+		// Make 500 retry attempts with 100ms wait time between each attempt to allow enough time for the lock to be released.
+		const retries = 500;
+		const retryWait = 100;
+
+		let retryAttempt = -1;
+		while (retryAttempt < retries) {
+			try {
+				// Use lockfile.lockSync() to ensure only one process is accessing the cache at a time.
+				// lockfile.lock() does not wait for async callback promise to resolve.
+				lockFile.lockSync(lockFilePath);
+				break;
+			} catch (e) {
+				if (retryAttempt === retries) {
```

Comment:
This will currently never be hit - you don't increase retryAttempt until after this check and it'll exit the while loop as soon as retryAttemp === retries